tests/functional_tests/scoreboard.bats

```diff
@@ -13,41 +15,93 @@ teardown() {
 # in all cases, check for equality of json objects received on socket.io-client. Using a modified submit.js would be appropriate
 
 @test "empty scoreboard of a fresh lab" {
-	# check for empty scoreboard
-	skip "TODO"
+	mkdir -p $BATS_TMPDIR/scoreboard/fresh
+	curl -s --ipv4 -k https://127.0.0.1:9000/scoreboard/lab1 -o $BATS_TMPDIR/scoreboard/fresh/empty.json
+	cmp data/scoreboard/empty.json $BATS_TMPDIR/scoreboard/fresh/empty.json
+	result=$?
+	rm -rf $BATS_TMPDIR/scoreboard
+	[ "$result" -eq 0 ]
 }
 
 @test "empty scoreboard of an inactive lab" {
-	# check for empty scoreboard after one evaluation
-	skip "TODO"
+	node submit.js -i 2015A7PS006G -l lab3 --lang=java --host='localhost:9000' >/dev/null
+	mkdir -p $BATS_TMPDIR/scoreboard/inactive
+	curl -s --ipv4 -k https://127.0.0.1:9000/scoreboard/lab3 -o $BATS_TMPDIR/scoreboard/inactive/empty.json
+	cmp data/scoreboard/empty.json $BATS_TMPDIR/scoreboard/inactive/empty.json
+	result=$?
+	rm -rf $BATS_TMPDIR/scoreboard
+	[ "$result" -eq 0 ]
 }
 
 @test "scoreboard after one evaluation" {
-	# check for scoreboard with one correct entry
-	skip "TODO"
+	node submit.js -i 2015A7PS006G -l lab1 --lang=java --host='localhost:9000' >/dev/null
+	mkdir -p $BATS_TMPDIR/scoreboard/first
+	curl -s --ipv4 -k https://127.0.0.1:9000/scoreboard/lab1 -o $BATS_TMPDIR/scoreboard/first/first_eval.json
+	cmp data/scoreboard/first_eval.json <(jq [.[].id_no,.[].score] $BATS_TMPDIR/scoreboard/first/first_eval.json)
+	result=$?
+	rm -rf $BATS_TMPDIR/scoreboard
+	[ "$result" -eq 0 ]
 }
 
 @test "scoreboard after two evaluations" {
-	# check for empty scoreboard with two correct entries
-	skip "TODO"
+	node submit.js -i 2015A7PS006G -l lab1 --lang=java --host='localhost:9000' >/dev/null
+	node submit.js -i 2015A7PS066G -l lab1 --lang=java --host='localhost:9000' >/dev/null
+	mkdir -p $BATS_TMPDIR/scoreboard/second
+	curl -s --ipv4 -k https://127.0.0.1:9000/scoreboard/lab1 -o $BATS_TMPDIR/scoreboard/second/second_eval.json
+	cmp data/scoreboard/second_eval.json <(jq [.[].id_no,.[].score] $BATS_TMPDIR/scoreboard/second/second_eval.json)
+	result=$?
+	rm -rf $BATS_TMPDIR/scoreboard
+	[ "$result" -eq 0 ]
 }
 
 @test "no change on scoreboard for worse score" {
-		# check for the correctly updated scoreboard
-		skip "TODO"
+	node submit.js -i 2015A7PS006G -l lab1 --lang=java --host='localhost:9000' >/dev/null
+	bash ./helper_scripts/scoreboard/worse_score1.sh
```

Comment:
appropriate file name changes for worse_score1.sh and worse_score2.sh as suggested before.